### ISSUE TITLE
PHP Warning:  Attempt to read property...

### ISSUE THREAD

**Issue by shmaltz** [soflyy/oxygen-bugs-and-features#3131]
Getting these PHP warnings using Oxygen v4.1.1 and PHP v8.0

```
PHP Warning:  Attempt to read property "post_type" on null in /htdocs/wp-content/plugins/oxygen/component-framework/admin/admin.php on line 313
PHP Warning:  Attempt to read property "ID" on null in /htdocs/wp-content/plugins/oxygen/component-framework/admin/pages.php on line 1581
PHP Warning:  Attempt to read property "ID" on null in /htdocs/wp-content/plugins/oxygen/component-framework/admin/admin.php on line 312
```

**Comment by Spellhammer** [comment]
What steps do we need to take to trigger these errors?

**Comment by shmaltz** [comment]
I'm trying to track down the cause of those errors, but here are some more errors I get when loading a single post template with an easy-posts element in it, in the Oxygen backend:

```
PHP Notice:  ob_end_flush(): Failed to send buffer of zlib output compression (3) in /htdocs/wp-includes/functions.php on line 5279
PHP Notice:  ob_end_flush(): Failed to send buffer of zlib output compression (3) in /htdocs/wp-includes/functions.php on line 5279
PHP Notice:  ob_end_flush(): Failed to send buffer of zlib output compression (3) in /htdocs/wp-includes/functions.php on line 5279
PHP Notice:  ob_end_flush(): Failed to send buffer of zlib output compression (3) in /htdocs/wp-includes/functions.php on line 5279
PHP Warning:  Undefined variable $old_query in /htdocs/wp-content/plugins/oxygen/component-framework/components/classes/image.class.php on line 209
PHP Warning:  Undefined array key "posts_per_page" in /htdocs/wp-content/plugins/oxygen/component-framework/components/layouts/easy-posts.php on line 48
PHP Warning:  Undefined array key "query_post_ids" in /htdocs/wp-content/plugins/oxygen/component-framework/components/layouts/easy-posts.php on line 48
PHP Warning:  Undefined array key "query_order_by" in /htdocs/wp-content/plugins/oxygen/component-framework/components/layouts/easy-posts.php on line 48
PHP Warning:  Undefined array key "query_all_posts" in /htdocs/wp-content/plugins/oxygen/component-framework/components/layouts/easy-posts.php on line 48
PHP Warning:  Undefined array key "query_order" in /htdocs/wp-content/plugins/oxygen/component-framework/components/layouts/easy-posts.php on line 48
PHP Warning:  Undefined array key "query_taxonomies_any" in /htdocs/wp-content/plugins/oxygen/component-framework/components/layouts/easy-posts.php on line 48
PHP Warning:  Undefined array key "query_taxonomies_all" in /htdocs/wp-content/plugins/oxygen/component-framework/components/layouts/easy-posts.php on line 48
PHP Warning:  Undefined array key "query_authors" in /htdocs/wp-content/plugins/oxygen/component-framework/components/layouts/easy-posts.php on line 48
```

**Comment by leolll** [comment]
I get similar PHP warnings when I use the Repeater component.  It happened when you are in the oxygen builder on a page with a Repeater.  Below is an example that produces warnings because the `if` statement assumes a key exists.

```php
// post type
if ($options['query_post_ids']) {
    $args['post__in'] = explode(",",$options['query_post_ids']);
    $args['post_type'] = $options['query_post_types'];
}
```

> /wp-content/plugins/oxygen/component-framework/components/classes/dynamic-list.class.php in Oxygen_VSB_Dynamic_List::setQuery at line 1076

The `if` condition should instead be this:
```php
if (array_key_exists('query_post_ids', $options)) {
```
Below are a few other undefined keys that are accessed in a similar way:
```
Warning: Undefined array key "query_post_ids"
Warning: Undefined array key "acfRepeaterFields"
Warning: Undefined array key "preview_post_id"
Warning: Undefined array key "is_shortcode"
Warning: Undefined array key "query_count"
Warning: Undefined array key "query_order_by"
```

**Comment by leolll** [comment]
> What steps do we need to take to trigger these errors?

Here are the minimum steps needed to reproduce these warnings (make sure PHP warnings are being logged):

### Repeater component
Edit a post with Oxygen. Add a Repeater and fill the minimum fields for a Query (ie select a Post Type that has posts).  Click "Apply Query Params".
```
PHP Warning:  Undefined array key "query_post_ids" in /wp-content/plugins/oxygen/component-framework/components/classes/dynamic-list.class.php on line 1076
PHP Warning:  Undefined array key "query_taxonomies_any" in /wp-content/plugins/oxygen/component-framework/components/classes/dynamic-list.class.php on line 1085
PHP Warning:  Undefined array key "query_taxonomies_all" in /wp-content/plugins/oxygen/component-framework/components/classes/dynamic-list.class.php on line 1109
PHP Warning:  Undefined array key "query_authors" in /wp-content/plugins/oxygen/component-framework/components/classes/dynamic-list.class.php on line 1134
PHP Warning:  Undefined array key "query_order" in /wp-content/plugins/oxygen/component-framework/components/classes/dynamic-list.class.php on line 1139
PHP Warning:  Undefined array key "query_order_by" in /wp-content/plugins/oxygen/component-framework/components/classes/dynamic-list.class.php on line 1140
PHP Warning:  Undefined array key "query_count" in /wp-content/plugins/oxygen/component-framework/components/classes/dynamic-list.class.php on line 1146
```
### Easy Posts component
Edit a post with Oxygen.  Add an Easy Post component.
```
PHP Warning:  Undefined array key "posts_per_page" in /wp-content/plugins/oxygen/component-framework/components/layouts/easy-posts.php on line 48
PHP Warning:  Undefined array key "query_post_ids" in /wp-content/plugins/oxygen/component-framework/components/layouts/easy-posts.php on line 48
PHP Warning:  Undefined array key "query_order_by" in /wp-content/plugins/oxygen/component-framework/components/layouts/easy-posts.php on line 48
PHP Warning:  Undefined array key "query_count" in /wp-content/plugins/oxygen/component-framework/components/layouts/easy-posts.php on line 48
PHP Warning:  Undefined array key "query_all_posts" in /wp-content/plugins/oxygen/component-framework/components/layouts/easy-posts.php on line 48
PHP Warning:  Undefined array key "query_order" in /wp-content/plugins/oxygen/component-framework/components/layouts/easy-posts.php on line 48
PHP Warning:  Undefined array key "query_post_types" in /wp-content/plugins/oxygen/component-framework/components/layouts/easy-posts.php on line 48
PHP Warning:  Undefined array key "query_taxonomies_any" in /wp-content/plugins/oxygen/component-framework/components/layouts/easy-posts.php on line 48
PHP Warning:  Undefined array key "query_taxonomies_all" in /wp-content/plugins/oxygen/component-framework/components/layouts/easy-posts.php on line 48
PHP Warning:  Undefined array key "query_authors" in /wp-content/plugins/oxygen/component-framework/components/layouts/easy-posts.php on line 48
PHP Warning:  Undefined array key "posts_per_page" in /wp-content/plugins/oxygen/component-framework/components/layouts/easy-posts.php on line 48
PHP Warning:  Undefined array key "query_post_ids" in /wp-content/plugins/oxygen/component-framework/components/layouts/easy-posts.php on line 48
PHP Warning:  Undefined array key "query_order_by" in /wp-content/plugins/oxygen/component-framework/components/layouts/easy-posts.php on line 48
PHP Warning:  Undefined array key "query_count" in /wp-content/plugins/oxygen/component-framework/components/layouts/easy-posts.php on line 48
PHP Warning:  Undefined array key "query_all_posts" in /wp-content/plugins/oxygen/component-framework/components/layouts/easy-posts.php on line 48
PHP Warning:  Undefined array key "query_order" in /wp-content/plugins/oxygen/component-framework/components/layouts/easy-posts.php on line 48
PHP Warning:  Undefined array key "query_post_types" in /wp-content/plugins/oxygen/component-framework/components/layouts/easy-posts.php on line 48
PHP Warning:  Undefined array key "query_taxonomies_any" in /wp-content/plugins/oxygen/component-framework/components/layouts/easy-posts.php on line 48
PHP Warning:  Undefined array key "query_taxonomies_all" in /wp-content/plugins/oxygen/component-framework/components/layouts/easy-posts.php on line 48
PHP Warning:  Undefined array key "query_authors" in /wp-content/plugins/oxygen/component-framework/components/layouts/easy-posts.php on line 48
```

**Comment by Kpudlo** [comment]
Thank you for the information. Reported internally via https://github.com/soflyy/oxygen/issues/4871.

**Comment by pioneer-ma** [comment]
Hey guys. Any updates on this issue?

**Comment by Kpudlo** [comment]
Hi @pioneer-ma, 

These should be fixed in Oxygen 4.6, which is available for Release Candidate testing. You are welcome to download and test Oxygen 4.6 RC 1 via the Customer Portal: https://oxygenbuilder.com/portal/.

**Comment by pioneer-ma** [comment]
> Hi @pioneer-ma,
> 
> These should be fixed in Oxygen 4.6, which is available for Release Candidate testing. You are welcome to download and test Oxygen 4.6 RC 1 via the Customer Portal: https://oxygenbuilder.com/portal/.

Got it! Will try to test right now. Thx)

**Comment by Kpudlo** [comment]
Fixed in Oxygen 4.6.
